### DISCUSSION TITLE
Data Object For Array

### DISCUSSION THREAD

**Discussion by hirenkeraliya** [spatie/laravel-data#36] [Q&A]
What do we do when we have an array on the request object?

```
'items' => 'nullable|array',
'items.*.name' => 'required|string',
'items.*.age' => 'nullable|integer',
'items.*.school' => 'nullable|string',
```

**Comment by rubenvanassche** [comment]
You would create a DataCollection with Data objects, the rules of that data object will be used as rules for the array.

**Reply by hirenkeraliya** [reply]
I see the documentation and I got your answer. But, I am new to this data object flow can you able to share the example for this one?

**Reply by rubenvanassche** [reply]
I've added a short section within the docs that explains it a bit: https://github.com/spatie/laravel-data/blob/b1286151c113edf30c894f1038b4e5a8b1ad5cf4/docs/as-a-data-transfer-object/request-to-data-object.md#L287

**Reply by Wulfheart** [reply]
@rubenvanassche And how would you validate then something like this:
```php
'no_adjudication' => ['array'],
'no_adjudication.*' => ['boolean', 'accepted'],
```

It seems I can't get my around this...

**Reply by rubenvanassche** [reply]
Take a look at the `rules()` method, you can manually define rules for each data object over there. I think a specific attribute would be something cool in the future to allow this.

**Comment by hirenkeraliya** [comment]
Anyone able to share the example of the crud with the file upload, normal string, array field with some optional and some required fields.

**Comment by hirenkeraliya** [comment]
One more thing I need your suggestion for is the Laravel data class file.

where I can put the file it's in the `app\Data\CartData.php`  or do I need to put this in another folder?

Currently, I put this on the `app\Domains\Sales\Data\CartData.php` is this fine?

**Reply by rubenvanassche** [reply]
You're free to put them anywhere you want 😄
